### DISCUSSION TITLE
Reconsider new subscription model and prices

### DISCUSSION THREAD

**Discussion by scaidermern** [osmandapp/OsmAnd#12187] [General]
It looks like there are new subscription models with a huge price increase, especially for (hourly) live updates:

![photo_2021-07-21_07-38-42](https://user-images.githubusercontent.com/1040784/126436948-6c5a0b64-9efa-4c64-b98f-2d0c0e78ce95.jpg)

I guess you have valid reasons for this price increase since infrastructure and labor won't come for free. However please consider that many mappers "need" the OsmAnd live feature. It allows them to efficiently contribute to OpenStreetMap by having up-to-date map data. This avoids creating duplicated POIs and enables mappers to instantly differentiate between current and outdated information.

While most mappers "need" OsmAnd live, many of them won't need the OsmAnd cloud feature. The OsmAnd cloud feature is probably the one which contributed to this price increase I guess.

Please take into account that active mappers are the one big reason why OpenStreetMap and OsmAnd are a big success. Without them, OsmAnd wouldn't exist and won't survive. Give mappers a chance to use your live feature without having to pay a huge amount of money. Otherwise there is the possibility that map edits and map quality will decrease, thus making OsmAnd less useful.

There are several options to solve this conflict:

- Move the hourly map update feature from the pro version to the plus version
- Alternatively, reduce the price for the pro version
- Alternatively, offer a third subscription model including the plus features and the live update feature
- Alternatively, offer a discount or free live update feature for active mappers (mappers with a certain number of edits per year or a certain number of active mapping days per year)

**Comment by DP24PH** [comment]
As a mapper myself and Civil Engineering student, I would like to reconsider the option to offer either discounted or free live update feature since it would be only the way to validate the edit in a timely manner. Likewise, it would rather incentivize the active mappers' contribution as well to maintain the quality of the OSM-based database.

**Comment by vshcherb** [comment]
Thank you for feedback. Indeed the things are not set in stone yet.

- The main aspect of this change to standardize offer solution and set it for next 5 years. You've correctly mentioned 2 products: OsmAnd Pro (only subscription) & Maps+.

- Main aspect: we don't want to introduce 3rd package.

- Indeed we need to introduce OsmAnd live package for mappers and probably make it available even without Maps+. For the time being we've considered F-Droid version for mappers cause it's available for free. Though some mappers still would like to support OsmAnd development for example.

- We can't provide OsmAnd Live with Maps+ package cause it's available as 1 time payment (that costs 2013-2017 only 5 EUR). 

As a summary I see:
- Provide a reasonable subscription of (7-10 EUR per month), so more people could afford supporting OsmAnd development (even if they are fine with OsmAnd+).
- Provide a solution that **active** mappers could have unlimited Maps & Live Maps download. That could be measured by user monthly / yearly activity somehow.

**Reply by HikeAndMap** [reply]
To generate more income, while not hindering mappers on low budget..

and at the same time people having the advantage of OSM generally as well as OSMAnd
while not contributing to it. 

(just thinking out loud here since I don't know what's possible in google play store)
but there's in-app purchases right..

something like:
1) specific amount of routing/directions/navigation are included in  the basic package
2) if you use navigation/directions more often you need to buy an additional subscription for that.

In NL we have this saying:"the gebruiker betaald" which basically means - the more you
use a service, the more you pay for that service.

It's about fairness - there are people that mainly contribute to OSM (like myself - for me it's 
mostly hobby really - I like doing it - although I rarely really use OSM since obviously most is
already in my head anyway and everything I add - well that's then in my head too LOL

there are people who do both contribute and use OSM 

there are people who don't contribute at all and use the work of programmers of OSMAnd
and the volunteers.. 

I think it should be reflected in the pricing model - these differences.

And although I'm using the f-droid - yes I payed still for the software on google play store
and the subscription every year - again out of fairness for the developers .

I would expect as mapper also a fairness towards mappers. And I'm not asking this for myself ,
as I'll just continue the yearly subscription anyway although by now I use JOSM and OSM Map creator
to create my own obf from the pbf so I get updated routes and public transportation instantly 
since these are not included in live updates sadly..

so I don't even use these services yet out of fairness and the support of OSMAnd still
paying in the store.

But the fairness is I am living in a country where a lot of poor people live, I mean really poor as in
some live from 50 USD a month with a whole family. And I know here a few people that contribute
to OSM like me as hobby - for them even 1 USD is food or not for the whole family for a day.

Such mappers shouldn't be discouraged from using OSMAnd ..

That's just my idea.. again, not for me, I'm good with pricing models whatever..
But the thing is - I wish more people here were contributing - as this country is very badly
represented in OSM still.. surely we don't wish to discourage mappers even further..

Just some thoughts here on my side of the world.

**Reply by scaidermern** [reply]
> something like:
> 
>     1. specific amount of routing/directions/navigation are included in  the basic package
> 
>     2. if you use navigation/directions more often you need to buy an additional subscription for that.
> 
> 
> In NL we have this saying:"the gebruiker betaald" which basically means - the more you
> use a service, the more you pay for that service.

This approach is not an option for an app targeted at offline usage. Also, route calculation happens on the device and not on the OsmAnd server. So using OsmAnd more often doesn't increase the cost for running the OsmAnd server infrastructure.

Daily and hourly live updates are a different thing, though. They need a good server infrastructure and are probably rather expensive to maintain. It is perfectly fine to charge more for this feature. However there should be an alternative solution for active mappers, especially in countries with low income.

**Reply by thompehw** [reply]
@vshcherb 
> * Provide a solution that **active** mappers could have unlimited Maps & Live Maps download. That could be measured by user monthly / yearly activity somehow.

Sorry if everyone knows this, but there is  a a precedent: active mappers can join OSMF for free: [https://join.osmfoundation.org/active-contributor-membership/](https://join.osmfoundation.org/active-contributor-membership/) 

Again, it's self-interest not charity: if OSMF has a lot of members who are genuine mappers then it is more resistant to takeovers.

**Comment by vshcherb** [comment]
I would summarize again based on different opinions.
- OsmAnd Live updates will be included in OsmAnd Pro.
- For OSM active mappers (active means at least 3-10 changes for previous 2 months) OsmAnd Live will be available for Free! Somehow, solution is being developed.
- OsmAnd Pro will include Backup plan and some newly developed features, so OsmAnd Pro will not be cheap cause it has direct cost on OsmAnd Cloud
- If user wants to support OsmAnd development but doesn't need OsmAnd Cloud (Pro), he could purchase Maps+ year subscription.
- Eventually all features (advanced widgets / features) except (OsmAnd Live & OsmAnd Pro) will be migrated to Maps+ / OsmAnd+ package. But there will be always small difference in functionality between Pro & Maps+ (we approximate that new features will be exclusive for Pro for about 3-12 months).

Feedback is welcome again.

**Reply by scaidermern** [reply]
Sounds good!

**Reply by scaidermern** [reply]
If you don't want to implement the "active contributor" query yourself: Maybe you can talk to Pascal Neis and ask if it is possible to get this data from [How did you contribute?](https://hdyc.neis-one.org) via an API or periodic dumps.

**Reply by vshcherb** [reply]
We have our own computation, you can check validity of it on https://osmbtc.org/

**Reply by HikeAndMap** [reply]
I like this concept .. especially the changes per 2 months - stay active  is a good concept

Anyway - I'll continue my yearly subscriptions regardless of the fact I don't need it simply because I love OSMAnd and the concept..

I wish - thought unrelated from OSMAnd - OSM had something in place like:" for the top 3 unmapped countries the top 3 contributors of that year get some award in form of financial support but not direct cash..

more like:"good mapping stuff, hardware for geotagging, etc.."

Again, this is not related to OSMAnd so not sure why I'm writing it here LOL

But I live in a country with a lot of poor folks and wish to somehow promote among them:"even if you're poor, instead of wasting your time playing games on the phone - contribute to the global good"

**Comment by vshcherb** [comment]
Please try Mappers benefit in OpenStreetMap Editing Plugin 
<table style="width:100%">
  <tr>
    <th><img src="https://user-images.githubusercontent.com/1042025/127700486-64ebe22b-64ef-45d5-a322-92c168f059ad.jpg"   width="300" height="auto"/>
</th>
    <th><img src="https://user-images.githubusercontent.com/1042025/127700528-0f944965-5555-41d3-8f4f-b86e6a7a7eb7.jpg"   width="300" height="auto"/>
</th>
  </tr>
</table>

**Reply by HikeAndMap** [reply]
Maybe 10 for 30 days is a bit too much.. we don't want people to make a change - commit, change commit, change commit..

I usually do a lot of changes of 4-5 hours work in a single commit and I'm sure I'm not alone in that. 

What you'd get then is people making cheap/small commits instead of quality commits..

or people if they done some commits 10 and still have changes.. they save them for next month(s)

a larger timespan would prevent that behavior.

**Reply by vshcherb** [reply]
it's about map object changes - not changesets.

**Reply by HikeAndMap** [reply]
Oh I mixed those 2 .. my mistake.. thanks for clarifying that

then I change my opinion into: 10 changes a month is a good motivation! :)

**Reply by scaidermern** [reply]
"Map updates for mappers" works for me (after hitting the refresh button) :+1:

**Reply by pebogufi** [reply]
I am typically mapping nearly only "outside", when traveling, making bike or hiking tours. In winter there is only very few mapping possible in new regions. So currently i have enough points in summer (June>100, July >600) but in winter i am only very few traveling.  I think also a minimum average over some (3?) months might be interesting for such passive periods where a mapper is not "on the road" to block him not in this time from getting updates.

**Reply by Zahnstocher** [reply]
@pebogufi This shouldn't be an issue, your lowest number of object changes within 2 months was 59 (since 2019). :smile:

**Reply by pebogufi** [reply]
@Zahnstocher  I know about statistics of http://hdyc.neis-one.org, but where did you get this numbers (59/2months) from ?
I would have expected big gaps in mapping over winter.

**Reply by Zahnstocher** [reply]
@pebogufi Please have a look at objectChanges here: https://osmand.net/changesets/user-changes?name=pebogufi

**Reply by pebogufi** [reply]
Danke !

**Reply by HikeAndMap** [reply]
Well to be honest - you can even map trash bins or street lamps or even poles..

getting 30 changes per month even in heavily mapped areas should still be doable.

And IF eventually an area is truly 100% mapped - and we're not contributing anymore therefore - it's only fair to say:'then you gotta pay for the software' because that's still being maintained and updated regularly

**Reply by Lee-Carre** [reply]
> you can even map trash bins or street lamps

As a contributor who does on-foot surveying (not armchair) in an area with not-the-best aerial imagery, I make a point of adding such small features which aren't seen in imagery. An important category is defibrillators, for example.

As one of a handful of local mappers on a small island, I find Live Updates invaluable (though I tend to limit it to daily). Thankyou @vshcherb & OsmAnd team for having the foresight to keep this feature available to those actively improving the people's map.

For example, outside of dedicated (multi-hour) surveying sessions, one way that I come to know of features which are absent or need updating, is via the ‘eat your own dog food’ method. That is, by using OSM exclusively (which is also out of reasons of principle; tech-freedom, network-autonomy, and such) I'm more likely to notice less-obvious problems (mistagging, questionable routing due to inaccurate map data, etc.).

> **if** eventually an area is truly 100% mapped

Then there's less need for frequent updates, since less will be changing 🙂. It's an interdependent feedback loop.

In such cases of there being little new to add or correct, then contributors are incentivised to be the first to notice or to seek out meat-space changes that aren't yet reflected in OSM data. Else, to resurvey volatile features (see StreetComplete's `Check if … still exists` quest, for example), confirm that the likes of opening_hours are still current. Shop vacancies will always be something to handle. Roads are resurfaced (`smoothness=*`), and so on.

Where I am, there is a vast amount to do for the few regulars (we're sorely lacking on even basics like GNSS traces). I welcome the day when there's only occasional updates needed (and I can set Live to weekly, or leave it disabled)!

**Reply by HikeAndMap** [reply]
I totally agree with you Lee-Carre

Indeed when everything that can be mapped 100% is done - then there's no need for regular updates indeed LOL

Anyway - here I'm in a 360k city, with around it another 600k people..
Over an area 30x30km

and I'm the only active mapper here. I indeed go out - walk around, make photos, videos, notes and that's how I map - with the aid of imagery..

But everything I do is backed by what I seen with my own eyes and made photos/videos/notes from.

I dare to say - even when 100 active mappers suddenly would come to the city and join me mapping everything in detail - it's going to take at least 10 years to get even close to the basics 100% mapped..

Not even thinking about all-details here.. 10+ years for 100 mappers to get the basics done.

It's an enormous task really - especially since the government quite often isn't forthcoming and as I have some contacts to local government - they quite often have even less data available than I already have collected..

it's gigantic the task here - and impossible for "armchair" mappers to do anyway.. 

So in a scenario where live updates matter - there's unlimited amount of work to do still

and as you said - if not much work to do left - then the live updates won't matter much anyway and even a once/year update would do I guess

**Reply by Lee-Carre** [reply]
@duiveldoder

> Anyway — here I'm in a 360k city, with around it another 600k people. Over an area 30×30km and I'm the only active mapper here.

Yikes. I thought Jersey was bad: ~108k population (& increasing rapidly), 45 square miles (yup; we have quite the housing crisis), ~2 locals contributing regularly (including myself).

Are you willing to identify which city? I'm curious to inspect it.

> I dare to say — even when 100 active mappers suddenly would come to the city and join me mapping everything in detail — it's going to take at least 10 years to get even close to the basics 100% mapped…
> 
> Not even thinking about all-details here… 10+ years for 100 mappers to get the basics done.
> 
> It's an enormous task really — especially since the government quite often isn't forthcoming and as I have some contacts to local government — they quite often have even less data available than I already have collected…
> 
> it's gigantic the task here — and impossible for "armchair" mappers to do anyway…

Oh, I feel your pain. Especially re government (on which I won't even get started; that would be a long & strongly-worded rant).

One reason I opted for surveying, is that about as much as can be done via armchair has already been done. Lots of buildings are missing, but landuse & highways are mostly done. The map is usable (for general navigation), at least. Rather than the mostly-blank canvas which it sounds like you have.

However, because much of that was done without GNSS traces to properly align imagery, there's a lot of misalignment (sometimes by several metres), which will need to be corrected in future. Makes placement of some features difficult; do I match to existing map-data, or GNSS position?(!)

Imagery is so lacking (with no prospect of improvement) that I've spoken with drone pilots about the possibility of them contributing to OAM. Trouble is, that one is never far from the airport (EGJJ), here, and the cost of the now-required certifications is, so I'm told, on the order of £2000 (but, they throw in *lunch*(!) 😄🤦‍♂️). That's on top of the price of equipment.

Aside from small features, a common effect of all the from-imagery mapping is that footpaths are marked as a straight-through path, since the resolution of imagery isn't sufficient to see where there are stairs. So much so that there are occasional notes added by accountless users, pointing out such snags (sometimes amusingly so; see [note #2890362](//www.openstreetmap.org/note/2890362), for example).

> even a once/year update would do

Well, the problem then would be that users may have to wait 11 months to see a change. So, monthly seems a good trade-off. However, if truly nothing changes month-to-month, then that might warrant a mechanism to inform clients (one possibility is to simply not publish new map files if they're no different to last month, or have a URL that always points to the latest version but which returns an HTTP 304 Not Modified unless there's an update, possibly also publishing (MD5) checksums for comparing versions).

**Reply by l29ah** [reply]
Tried it, it says that map updates are available to me, but i can't seem to find the way to actually download the maps: the map list still only shows maps that updated at march 1st.

**Reply by tommycrock** [reply]
@l29ah it usually takes about a week where I am for new maps to be generated at the start of the month. Areas with less data get new maps faster. The only way to get more up-to-date vector maps is with live maps, or generating your own using MapCreator.

**Reply by HikeAndMap** [reply]
Yes the City I'm talking about is Baguio.

Here's a link: [https://www.openstreetmap.org/#map=12/16.3959/120.6800&layers=ON](url)

And it's not just this city - I'm now going also to Agoo, Dagupan, Aringay and other places.. field surveying, or field mapping is really missing here - among the locals there's no contribution.

And to me that's sad, because you know I see people sitting doing nothing everywhere. They're playing TikTok, Video Games, etc it's rampant here like not just the young generation also the generation 20-40yo so many just sit around playing video games or doing TikTok it's really sad. They could be doing constructive work for the community - you know be busy and know you contributed to something in life other than tiktok and video games.

But I noticed most of them don't know about OpenStreetMap.

That's really sad - Grab, FoodPanda it's like half of the people use them for food services.
And these drivers used to call me always as my location wasn't know.

Then I update OSM and now a year later Grab updated the map finally - now they find me now my address is known to Grab.

But I can't put addresses of the whole city of course. Every now and then I add 50 to 200 more addresses.. But there's like 200k addresses LOL

But people don't know!!!!! Seriously 100.00% of the people I talk to have no idea nor ever heard about OSM!!

Also facebook - that's used here 10 times more than in Europe. Literally everyone uses facebook.

When you do messenger, which maybe dunno 90% of the people here seem to use, at least among the 0 to 40yo - location services of messenger is 100% Openstreetmap.

When you do shared location - Openstreetmap

When you search for shops on marketplace - all location services is openstreetmap

Health center and government agencies are all the straight forward way to communicate here with local government:"facebook" and all their services are --- Openstreetmap..

Every time you click on the "i" in Facebook or Grab or FoodPanda or Mercury Pharmacies, etc.. it's always "openstreetmap"

yet 0.00% of the people I talked with, which by now must be over 1000 folks - not a single person ever heard about OpenStreetMap

So my guess is - all these TikTok and Video Gamers - if they knew - some would contribute right.. but the problem is really awareness.. no one is aware that we as civilians can contribute.

That's the sad reality here.. there's no promotion for OSM - although it's really necessary here.

And again - I don't blame the people.. nor do I wish to replace the "i" with something else.

But Facebook, Grab, etc might consider an active promotional campaign for OpenStreetMap at least in a country like there where >90% of the local population aged 0 to 40 is not just using Facebook, they're relying on it as one of the main channels to communicate among each other, shops, establishments, governmental offices, etc..

**Comment by thompehw** [comment]
I noticed that the "available until" date of map updates for mappers (in the OSM editing plugin) wasn't changing and wondered how it was calculated. From what I have found from searching the code repository (I'm no expert) the current date of 16 Feb is just written into the code.

Does anyone know about the long-term intentions? It's sometimes called a "mappers' benefit" but in the code "PROMO" is also used.

Personally, I prefer this as a way of rewarding mappers than the bitcoin scheme (which I would never consider subscribing to), but I understand that OsmAnd must be careful to understand the financial effects.

**Reply by Lee-Carre** [reply]
Perhaps the algorithm is a work-in-progress, and the hardcoded date will be bumped every so often?

Unless it's to test behaviour of when the date is exceeded.

**Reply by Lee-Carre** [reply]
Noticed, today, that mine now reads `16 Mar`, so if it was hardcoded, it isn't anymore.

Are you sure that you didn't just find the strings which are (conditionally) output? A link to where in the code you were looking, would help.

**Reply by thompehw** [reply]
Yes, mine has incremented by a month too. Hooray! Thanks, OsmAnd devs!

@Lee-Carre this is just out of casual interest now as there is clearly nothing to worry about, but I was looking in OsmAnd/OsmAnd/src/net/osmand/plus/plugins/osmedit/fragments/MappersFragment.java

I know C so I can usually guess what Java means, but I now guess that checkLastChanges increments the month by 1 (with the rest of the date hard-coded) rather than the "1" signifying February! 

Just out of curiosity, I'd be interested to know what condition triggers the increment - I haven't spotted it.

Best wishes.

**Reply by HikeAndMap** [reply]
It's not hardcoded..

I'm not really sure how it's coded but there's many ways to have a date being coded into an algorithm.

One of them is to actively use the data on the clients phone to set a date - then I guess the date would update every day as your date passes by.

But another way is to code it "semi-hard coded" where you every month when the generation of the map is - use that generation data date - also as a date where you pull the latest updates on the map - and then you add +2 months there

another way is - in visual studio for instance you can "semi-hard-coded" rely on the date where you compile a new version and then add +2 months (and every time round up to a specific date in the middle of that month)

and there's tons of things I can think about.

But basically..

hard coded would be every time I compile I write something like
var datetimestring= "20220616T00:00:00Z"; in C-sharp code
DateTime iso8602 = DateTime.ParseExact(datetimestring, "yyyyMMddTHH:mm:ssZ", System.Globalization.CultureInfo.InvariantCulture);
And then I update the datetimestring manually - and hope I won't forget it.

Another way, in C Sharp, would be the new constructor like: new DateTime(2022, 6, 16, 00, 00, 00);

Both ways then you have your extension period until the next 2 months June 16 2022 (there's other means too, many to do this actually)

I'd say no programmer in his right mind would do that for this functionality as "forgetting" tends to be a given fact will happen LOL

Then upon runtime (not hard coded): You skip the datetimestring and parsing it completely, the code looks leaner
DateTime dt = DateTime.Now;
dt.AddMonths(2);

Here the code might become complicated to deal with the fact people can fake datetime on their equipment.. Programmatically you can deal with it - but it's extra work to cover all the possibilities how users can fake datetime on their equipment.

And then there's the "semi hard coded" - my guess is this they went for. Now I don't know the exact syntax for android or ios programming, but again in C-Sharp it would look like this..

var version = Assembly.GetExecutingAssembly().GetName().Version;
DateTime buildDate = new DateTime(2000, 1, 1).AddDays(version.Build).AddSeconds(version.Revision*2);
    
Now it doesn't matter what the user sets the date or time on his equipment, it's now "hard coded" if you wish but automatically changes every month you compile a new version of OSMAnd..
    
just do the
buildDate.AddMonths(2)  and you have your "grace" period for active mappers..
    
And now you can skip the work for all the coding to detect tampering of date time for a user to bypass the limitation by tweaking the data on their equipment..
    
So that's what I call the "semi-hard-coded" it's hard coded - but not really..
    
A more correct way would be to say:
it's hard coded variable
as opposed to 
hard coded static
    
but then again, I have to emphasize here the word "more" correct because - I didn't define the string as static either so that's technically also not entirely correct. 
    
Thus I rather use the description "semi-hard-coded" 
    
And my guess is - the latter option is what the devs of OSMAnd have chosen.. the "semi" approach.
    
As would I have chosen that option to avoid the extra coding to try to evaluate if users are tempering with date on their equipment..

**Comment by Lee-Carre** [comment]
Another factor to consider is if a mapper has applied to receive cryptocurrency (Bitcoin, or whatever it may be in future) from OsmAnd for their contributions to OSM.

I've deliberately not done so, in order that OsmAnd keep more (or it goes to mappers who need it). In some sense, this could be considered an indirect donation.

**Reply by HikeAndMap** [reply]
Yeah ditto - I map a lot - but skip any donations. There's very poor people out there and I wish them to map their areas too - areas I wouldn't necessarily go to.. giving 10 USD to a poor person goes much further than giving 10 USD to a person already making >10k a month

**Reply by rugk** [reply]
Is that cryptocurrency payout to contributors even still live? If I go to https://osmand.net/osm_live this redirects to https://osmbtc.org/, which has a certificate error and even if that is dismissed for testing, it just serves the usual OSM website.

**Reply by rugk** [reply]
Okay opened https://github.com/osmandapp/web/issues/310 for that.
